lib/ash/resource/transformers/validate_accept.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Ash.Resource.Transformers.ValidateAccept do
+  @moduledoc "Validates that accept and reject lists only contains valid attributes"
+  use Spark.Dsl.Transformer
+
+  alias Spark.Dsl.Transformer
+  alias Spark.Error.DslError
+
+  @impl true
+  def after_compile?, do: true
+
+  @impl true
+  def transform(dsl_state) do
+    {private_attributes, public_attributes} =
+      dsl_state
+      |> Transformer.get_entities([:attributes])
+      |> Enum.split_with(& &1.private?)
+
+    public_attribute_names = MapSet.new(public_attributes, & &1.name)
+    private_attribute_names = MapSet.new(private_attributes, & &1.name)
+
+    Transformer.get_entities(dsl_state, [:actions])
+    |> Enum.each(fn
+      %{name: action_name, accept: accept, reject: reject} ->
+        validate_attribute_name = fn attribute_name, type ->
+          cond do
+            MapSet.member?(private_attribute_names, attribute_name) ->
+              raise DslError,
+                path: [:actions, action_name, type, attribute_name],
+                message: "#{attribute_name} is a private attribute"
+
+            MapSet.member?(public_attribute_names, attribute_name) ->
+              :ok
+
+            true ->
+              raise DslError,
+                path: [:actions, action_name, type, attribute_name],
+                message: "#{attribute_name} is not an attribute"
+          end
+        end
+
+        if accept != :all do
```

Comment:
`accept` should never be `:all` here (same for `reject`), because an earlier transformer should remove it. Are you seeing it as `:all` at this point?

Reply:
I am, this test here is failing without those `if`s:
https://github.com/ash-project/ash/blob/14a1b92963beafe9d91586539b2cf5f3d89d73a1/test/resource/actions/actions_test.exs#L45-L65

Reply:
Looking further into it, it seems like the earlier transformer only updates `accept`, but not `reject`.
I assume `reject` is not needed after updating `accept`, so it isn't updated.